### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by mikiquantum**
Based on: 

- https://github.com/centrifuge/tinlake-erc20/blob/master/src/erc20.sol
- https://github.com/centrifuge/tinlake-erc20/pull/5/files
- https://github.com/centrifuge/radial/pull/1/files

### CODE REVIEW

File: src/test/permit_values [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
```
Comment (mikiquantum) [inline src/test/permit_values:1]: I believe needs to be adapted with new permit hash

File: src/test/wcfg.t.sol [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+    function testTypehash() public {
+        assertEq(wcfg.PERMIT_TYPEHASH(), 0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9);
+    }
+
+    function testDomain_Separator() public {
+        assertEq(wcfg.DOMAIN_SEPARATOR(), 0xd8c4cf6fb0cafceb796a57d02d0da83b5de9e840e36fe3e6558c8b9934eb62fc);
+    }
+
+    function testPermit() public {
```
Comment (mikiquantum) [inline src/test/wcfg.t.sol:290]: This test fails recovering addr based on v,r,s

File: src/test/wcfg.t.sol [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+        assertEq(wcfg.allowance(cal, del), 10000);
+        assertEq(wcfg.nonces(cal), 1);
+    }
+
+    function testFailPermitAddress0() public {
+        _v = 0;
+        wcfg.permit(address(0), del, 0, 0, v, r, s);
+    }
+
+    function testPermitWithExpiry() public {
```
Comment (mikiquantum) [inline src/test/wcfg.t.sol:303]: This test fails recovering addr based on v,r,s

File: src/wcfg.sol [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+  // --- Math ---
+  function add(uint x, uint y) internal pure returns (uint z) {
+    require((z = x + y) >= x, "math-add-overflow");
+  }
+  function sub(uint x, uint y) internal pure returns (uint z) {
+    require((z = x - y) <= x, "math-sub-underflow");
+  }
+
+  constructor(uint256 chainId_) public {
```
Comment (mikiquantum) [inline src/wcfg.sol]: Passing chainId in constructor

File: src/wcfg.sol [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    wards[msg.sender] = 1;
+    DOMAIN_SEPARATOR = keccak256(
+      abi.encode(
+        keccak256('EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)'),
+        keccak256(bytes(name)),
+        keccak256(bytes(version)),
```
Comment (mikiquantum) [inline src/wcfg.sol:62]: Using `constant version`

File: src/wcfg.sol [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    emit Transfer(usr, address(0), wcfg);
+  }
+  function approve(address usr, uint wcfg) external returns (bool) {
+    allowance[msg.sender][usr] = wcfg;
+    emit Approval(msg.sender, usr, wcfg);
+    return true;
+  }
+
+  // --- Alias ---
+  function burnFrom(address usr, uint wcfg) external {
```
Comment (mikiquantum) [inline src/wcfg.sol:108]: Added burnFrom alias to support BurnableMint tokens + ChainSafe logic

File: README.md [hunk 7]
```diff
@@ -1,2 +1,46 @@
 # WCFG ERC-20 Contract
 Ethereum Wrapped Centrifuge Token (wCFG)
+
```
Comment (lucasvo) [inline README.md:3]: I would delete all the radial specific parts below

File: src/lib.sol [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+contract MintLike {
```
Comment (lucasvo) [inline src/lib.sol]: where is this needed?

Comment (mikiquantum) [reply src/lib.sol]: no-where I presume, just leftover from previous code

**Comment by mikiquantum** [comment]
Closing in favor of using adapted contracts https://github.com/centrifuge/tinlake-erc20/blob/master/src/erc20.sol

**Pull request closed without merging**